Utilize TitanUtil's PathFinder

RyanHir opened the pull request:
Note: TitanUtil's PathFinder is a wrapper to WPI PathWeaver

I am adding @democat3457 as a reviewer as we will need more than one person to understand the changes in robot code, you don't have to fully understand the internals of the TitanUtil functions.

This code should not be merged until the team has a working demo on the physical robot.

RyanHir committed:
Use path_weaver snapshot for TitanUtil

RyanHir committed:
Add DrivebaseConfig

RyanHir committed:
Early first PathWeaver implementation

RyanHir committed:
Use new class name

RyanHir committed:
Use new custom Builder for pathweaver config

RyanHir committed:
Workspace configs

RyanHir committed:
Remove old 2020 implementation of PathWeaver configs

RyanHir committed:
misc

RyanHir committed:
Use new TitanUtil JoystickButton

RyanHir committed:
Fix build changes

RyanHir committed:
Use new TitanUtil POVButton

RyanHir committed:
Remove use of deprecated variable in TitanUtil

RyanHir committed:
Switch to TitanUtil's DrivebaseSubsystem

RyanHir committed:
Add proper deprecated flag

democat3457 committed:
remove .idea gen files

democat3457 commented on src/main/java/frc/robot/Constants.java:
```diff
@@ -126,34 +129,50 @@
+    // ================================================================================
+
+    public static final String DRIVEBASE_PATHWEAVER_PATH = "";
+    public static final DriveConfig DRIVEBASE_PATHWEAVER_CONFIG = new DriveConfig.Builder() //
```
Are we not going to use the other constants that were in the Auto/Drive Constant files?

RyanHir replied:
This will be discussed at the meeting

democat3457 commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -22,7 +21,6 @@
 
   @Override
   public void robotInit() {
-    Logger.DEBUG = true;
```
Why is `DEBUG` off?

RyanHir replied:
I deprecated `Logger.DEBUG` and is unused. I deprecated it as the logs can just be ignored during Comp so no reason to have a flag to disable it

democat3457 commented on src/main/java/frc/robot/subsystems/Drivebase.java:
```diff
@@ -181,11 +181,23 @@ public void setSlot(int slot) {
     }
 
+    /**
+     * @deprecated use driveTank
```
Why was this deprecated?

RyanHir replied:
There are now prewritten functions in the abstract class which do the same

democat3457 commented on src/main/java/frc/robot/subsystems/Drivebase.java:
```diff
@@ -222,23 +241,10 @@ public void setRamping(double ramping) {
         right.configClosedloopRamp(0);
     }
 
-    public double getRamping() {
```
Is this method just unused (and thus unnecessary)?

RyanHir replied:
It is unused IIRC and was there for our old MotionMagic code

democat3457 committed:
Fix comments

- Fixes RobotMap comments that got yeeted by the formatter
- Remove unused import, suppress unused warning

democat3457 committed:
Use undeprecated methods

democat3457 committed:
Reee formatting messes up comments

democat3457 committed:
More formatting

- Removes the unused method since it is reimplemented in WaitTillFlywheelAtSpeed
- Spacing changes

democat3457 committed:
oops fix unused var

RyanHir committed:
octocat

        `.  `,`,,:>|>|t+tsi?(?!+~:.`
	`.:>^||``\,^~``.|
	        _^,"!!|i+i)czclt}i!+^+il"|,`            `\\_:
		``.:,.  \>\`  `,>
		       `^|_:~\>?clzc!!i|^^!ttc)ct_^```,\|>+i)czc!|i+:
		       `    ,^:
		              ,::|>+!r!i+>~+i!r>!+vlc)>"icanCd00%%0%p0C5zzz+\,
			      `^czl!xAO+  }0|
			            ..,.:^|~~:,,..>\:+!+czeen80%gAANmm&DRRBB@Y00000qSo)|+s+rC00%OMDmc
				                .:,,,:,
						`,!raL0000%m@RMRBBRBHHHD&DBX%h%00SbTCCCdPAO&@DBBXmy
						                 `\}zC00000YPO@BBBRBHHHHHBm%g&HHRBm00cv000PWBHHHWkBHRWdi`
								                `i5w00$00000G@&BB@O@HHHHHHRPAXBHHBRRP0Ic0$0@HHHHM!`CRHB&gr`
										  !?>,.
										  ,}n00000000000O@RHMPA@MX&m$GM&B@k&ANBHMp0x$00W@RHR}|Cz,5mBHRh!`
										    l@OIwdTc~`,z000000000q00%@DRRBBMm)\)CndC\!cd?.cRHBm0q000POgBWr@Hk
										    `+mHBn
										       c&s``rg0elaq0000dnq0q0N&BB@D&mSr,^cS00C>z&pczqWHHDh0000%mmBB&G%s,.|pRml.`
										          `C@fz0AYqL%h%0000000$0X@RNSe)Iya800000qf%Mk000NBBO0000nC00ORHHHBD&@B0:
											      `nBMDBM&mWDWOmA00000ARXl,n$ccxpdly$0000I%00$$NBBP0000$00b8gmmGk%Agh+,
											          !XHHMP00$0%Pm%mP00$0ABMAf$A0$x%Cn0008C0LI0ftigBRgdd000000000Czccccc!`
												     `ABHDk0pAA0$%BMkG%000ABHBP0p$dT000000Ld0Cz+|>c0BRkCC00000000000C}~~`
												        `mHBA0GBHBO00MBP00000kRHBGbCq0000000000000psI0NBRkCC000000000dCaC0z`
													   `OHBmgA@@mp0$@BA00000PBHBP"\e00000000000$0P000XHRkdC000000000$0000z`
													      .0DMRBMXPghkGRRA0qL0%OBHBA8S0TnC0000000$00N000@H@%0b00000000000000t`
													         `ykm@BRRDBBBR@N00dS$$ABBBA0000qd0000000q00p0wyWH&00b00000000000APhe):
														     "S0%P&DMRM&m%$00q00$ABRRg0000000000zC000$0c_\WBG00S0000$gNgp0$a!,\sxn+
														      :^~_5NOGPgp%000000$$00$gRHBA000000000$Cw000$}`|xRHMA0$000000kAgIneeoc>`:\
														      zzccccci>|+zbhg%000000b00@HBA00000000000p0000I!SgBHBP0qqq%00wz+,`
														      ,+5+
														      r"_~_>?(czzzac}z000000C$0@HBk00000000000an00000PgBHRgwyCSXA0c
														      :|||||\,.
														      `rq0000CccpBBGt_n00000000wd0000%GgBHRxn50Lk&0I
														                      |d0000d!\AHHNrz000$00000000000gmgBHD!yarq%@S0,
																                      !00000Cl0&HHX00%0%0000000w0000gPgRHD!+a\x0Xx$!
																		      \i(
																		                      c0$000l!SGBHRbC00%0000$00y0000PexRHHc,I!i0Offz`
																				      ,no!|
																				                     .C0$00b>5CnBHHq_a0h0000$00d00000+.OHHc`cl\CgwzT>
																						     ya
																						                    \0q$00Iz$e:mHHM!n000h%k00qC00000C5@HHc
																								    !x,z00sdz`
																								    r8\
																								    ,
																								                   !0C000Sw0z`OHHBPk000kp0$$0$0000000@HHc
																										   ,I|+00zew"`
																										   >zcc
																										                 `v0y000000f!mPBB&pC00000$00q00005^zRHHc
																												 ?z~C0qv0n\
																												 !wq
																												               `zqa0$0000zzN!WRgdv+cT00$000SdqIx+0BHR!
																													       \I!x00ln0)
																													       !C\C
																													                     :I8y00000q!,LkhMme\`.zd0000I~|i,rOBHHBe
																															     `z(c00xc0I,
																															     iAacb
																															                   +0Tn00$w0n,
																																	   `}A&Bfoc|.\eC|+dCn0p@RRHBBN,
																																	   +as0$0lS$+
																																	   _5.
																																	   ,
																																	                `z0nf00xI0c`
																																			`:dNRPANkApAzaOmP&RRBBRPPRn
																																			,x+I00cz0a.
																																			             ~80r5$0fC$!
																																				     `,fzkRBDmDHRHHHHM@HOz+C00X@_
																																				     `ccv0$C+Sq+
																																				                 `+00!n00eqS^
																																						 .aI,r0OW&WGW@XNhns`
																																						 ,y0hm>
																																						 |I}d00ts0n:
																																						             `v0q!d0dz0f.
																																							     `e$>`
																																							     ``.a0t~``
																																							     `>L0q+`
																																							     ,x+c00z\I0(`
																																							                 `y0b?00fz0c`
																																									 l0x`
																																									 ^Cdx,
																																									 `(00n\
																																									 `cl~00C\}0n~
																																									             :C0xc00zn0t
																																										     r0x`
																																										     `z0q"`
																																										     ,f0$z``_n~n00+,y0c`
																																										                 |d0cy00zqd|
																																												 r0x`
																																												 +00s`
																																												 !w0d!``zsc00f,!d8|
																																												             +00ty0Lc0n:
																																													     r0n,
																																													     _d0z`
																																													     `z00f,`!a|n0w|.e0z`
																																													                 l0q!C0Iz0e`
																																															 !0w_
																																															 ,n0n,
																																															 :I00v``z_}00r`>Tq!`
																																															            `a0C|$0nI0?
																																																    >0w_
																																																    `z0q>
																																																    !L0d!`|z_S0n:`r0C\
																																																               \C0C|00xn0"
																																																	       >0w_
																																																	       }00t
																																																	       `l00x:`f+c00i`,e0o,
																																																	                  +$0C|00cI0"
																																																			  >0w_
																																																			  |T0y`
																																																			  ,500z.+y!d0e.
																																																			  :x0c`
																																																			             +00f:00?x0!
																																																				     >0w_
																																																				     \C0q!
																																																				     _C0$+,zvI0w!
																																																				     `!q8|
																																																				                i00a:00}z0y,
																																																						>0d_
																																																						`e00c
																																																						`z00z`!xa00e.
																																																						.y0c`
																																																						           i00a:00is$d|
																																																							   t08^
																																																							   `i00C`
																																																							   >d0w:,xzd00!
																																																							   `!qT|
																																																							              i00a:00(x00"
																																																								      r0x`
																																																								      ~S00:
																																																								      .e00+`(ef00c`
																																																								      .50c`
																																																								                 (00a:00zw00"
																																																										 r0x`
																																																										 `e00t
																																																										 "L0n\^no00C\
																																																										 +0n,
																																																										 :^_|_,`
																																																										 `c00z:00I000"
																																																										 r0x`
																																																										 >w0n,
																																																										 `c00s`+CT$0r``\C0!
																																																										 zzzzzcri!>\z$0v,00000w_
																																																										 r0f`
																																																										 .a00!
																																																										 |T0n^.5C00n,`.e0e
																																																										 zzzzzzzzzzzq00r+00000C:
																																																										 ``r0f`
																																																										 `r00v`
																																																										 `s00I:"bq00r`
																																																										 !08
																																																										 zzzzzzzzzza%00zI00000C\.~+})t50x`
																																																										 `.`
																																																										 >q0x,
																																																										 :000c,z$00C\
																																																										 .e0
																																																										 zzzzzzzzzzI%0wcb00000Csczzzzzn0C++!">">~|\,`.y00+.``,.````
																																																										 `e00q+~Cd00c.
																																																										 !$
																																																										 zzzzzzzzzzn%%Cx000000bzzzzzzzn%Lcvi+!+i)zcl?!z00(````..``
																																																										 `
																																																										 \n00C>zn$0d!.,f
																																																										 zzzzzzzzzzS00S$0000d0bzzzzzzzn%Tzzzzzzcrrsrlca00z\:`.,.:::+cc:"C00IrCn00Clla
																																																										 zzzzzzzzze%0000000000Czzzzzzzn%qzzzz5azzzzzzcz00nczclcceyxb%g+.t000Cdn00%xzz
																																																										 zzzzzzzzzy%000000000%nzzzzzzzn%%ezzzzezozzzzzz%00oeaISbnIzfAAircI0000CI00$zz
																																																										 zzzzzzzzzn00000000000nzzzzzzzn%Lzzzzzzzzzzzzzzd0%fzezzzzzzzzzzzzzf000%fT00Cz
																																																										 zzzzzzzzzn00000000000nzzzzzzcb0CzzzzzzzzzzzzzzC0%nzzzzzzzzzzzzzzzza000Lf000y
																																																										 cclsti+++cCCCaCLbn8TLoccczzzzq0nzzzzzzzzzzzzzzn00Clzzzzccczzzzzzzzza000dd00C
																																																										 !+++!|\~|leazeofzawqn(>_>"!+r8dcrvlcccccczczctsS0n!""""!"!+}}iitsvv(vn8qSd$C
																																																										 :^:,,,,,,\:,.!rzirzxz^```.`,\ze.,,,\~\:::\\\:..lTy\...`.,::\\\:,,,,.`_eSIzCa
																																																										              `\>_\^,`
																																																											      ``
																																																											      `,>|`
																																																											      :rz+>|
																																																											                                                                             .>~.

RyanHir committed:
Merge branch 'path_weaver' of github.com:frc5431/RobotCode2021 into path_weaver

democat3457 committed:
Question

The drivebase pathweaver path is hardcoded. Should we use environment variables instead?
